### PR TITLE
Use WFWX for station data

### PR DESCRIPTION
**Pull request by dgboss**
# Test Links:
[Landing Page](https://wps-pr-2858.apps.silver.devops.gov.bc.ca/)
[MoreCast 2.0](https://wps-pr-2858.apps.silver.devops.gov.bc.ca/morecast-2)
[Percentile Calculator](https://wps-pr-2858.apps.silver.devops.gov.bc.ca/percentile-calculator)
[MoreCast](https://wps-pr-2858.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-2858.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-2858.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-2858.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-2858.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-2858.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

**Review by andrea-williams — APPROVED** [review]

File: .github/workflows/deployment.yml [hunk 1]
```diff
@@ -143,7 +143,7 @@ jobs:
         shell: bash
         run: |
           oc login "${{ secrets.OPENSHIFT_CLUSTER }}" --token="${{ secrets.OC4_DEV_TOKEN }}"
-          MODULE_NAME=api SECOND_LEVEL_DOMAIN="apps.silver.devops.gov.bc.ca" VANITY_DOMAIN="${SUFFIX}-dev-psu.apps.silver.devops.gov.bc.ca" bash openshift/scripts/oc_deploy.sh ${SUFFIX} apply
+          MODULE_NAME=api SECOND_LEVEL_DOMAIN="apps.silver.devops.gov.bc.ca" VANITY_DOMAIN="${SUFFIX}-dev-psu.apps.silver.devops.gov.bc.ca" USE_WFWX="True" bash openshift/scripts/oc_deploy.sh ${SUFFIX} apply
```
Comment (conbrad) [inline .github/workflows/deployment.yml:146]: Do we need `USE_WFWX="True"` anymore?

File: api/app/jobs/common_model_fetchers.py [hunk 2]
```diff
@@ -169,10 +169,10 @@ class ModelValueProcessor:
     """ Iterate through model runs that have completed, and calculate the interpolated weather predictions.
     """
 
-    def __init__(self, session):
+    def __init__(self, session, station_source: StationSourceEnum = None):
```
Comment (conbrad) [inline api/app/jobs/common_model_fetchers.py:172]: What's the difference between `None` and `UNSPECIFIED` for station source? Can we simplify?

File: api/app/tests/weather_models/test_env_canada_rdps.py [hunk 3]
```diff
@@ -119,4 +120,4 @@ def test_process_rdps(mock_download,
     # All files, except one, are marked as already having been downloaded, so we expect one file to
     # be processed.
     sys.argv = ["argv", "RDPS"]
-    assert app.jobs.env_canada.process_models() == 1
+    assert app.jobs.env_canada.process_models(StationSourceEnum.LOCAL_STORAGE) == 1
```
Comment (conbrad) [inline api/app/tests/weather_models/test_env_canada_rdps.py:123]: Do we only use `LOCAL_STORAGE` source for tests? If so, would it make sense to rename it to reflect that?

File: .github/workflows/deployment.yml (hunk 1 above)
Comment (dgboss) [reply .github/workflows/deployment.yml:146]: Yes, still needed for deployment of our `wps-api` containers.

File: api/app/jobs/common_model_fetchers.py (hunk 2 above)
Comment (dgboss) [reply api/app/jobs/common_model_fetchers.py]: Good catch...should be `UNSPECIFIED`

Fixed in https://github.com/bcgov/wps/pull/2858/commits/24aa59521fb7628be4edb52f24560ccc36920504

File: api/app/tests/weather_models/test_env_canada_rdps.py (hunk 3 above)
Comment (dgboss) [reply api/app/tests/weather_models/test_env_canada_rdps.py:123]: At this point I think this is the case. We can probably rip out all the logic associated with using the WFWX API for station data and just always fetch the latest. We'd just have to mock the call for testing purposes.

Comment (dgboss) [reply api/app/tests/weather_models/test_env_canada_rdps.py]: I renamed `LOCAL_STORAGE` to `TEST` in https://github.com/bcgov/wps/pull/2858/commits/7f28f715ed9e44df76e1f88904722b176eb07ad3

File: .github/workflows/deployment.yml (hunk 1 above)
Comment (conbrad) [reply .github/workflows/deployment.yml:146]: Yeah looks like we still have a lot of flags in the codebase relying on it. I wonder if we should add a dev task to remove them, or keep them around as a fallback? Not sure.

**Review by conbrad — APPROVED** [review]

File: .github/workflows/deployment.yml (hunk 1 above)
Comment (dgboss) [reply .github/workflows/deployment.yml:146]: Created a ticket to address this: https://github.com/bcgov/wps/issues/2865

**Pull request merged**